test/src/core/util/featureGraph.spec.js

```diff
@@ -0,0 +1,89 @@
+
+
+/*****************************************************************************
+ * Copyright (c) 2021 Bosch.IO GmbH
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ ****************************************************************************/
+
+const FeatureGraph = require('../../../../src/core/util/featureGraph');
+const Logger = require('../../../../src/core/util/logger');
+
+describe('core.util.featureGraph', () => {
+    let fg = null;
+
+    beforeEach(() => {
+        fg = new FeatureGraph();
+        fg.logger.setLogLevel(Logger.__LOG_LEVEL.NONE);
+    });
+
+    it('should detect simple dependency cycle', () => {
+        fg.addDependency('1', '2');
+        fg.addDependency('2', '1');
+
+        expect(fg.containsCycles()).toBeTruthy();
+    });
+
+    it('should detect complex dependency cycle', () => {
+        fg.addDependency('1', '2');
+        fg.addDependency('1', '4');
+        fg.addDependency('4', '5');
+        fg.addDependency('5', '8');
+        fg.addDependency('4', '9');
+        fg.addDependency('9', '5');
+        fg.addDependency('8', '4');
+
+        expect(fg.containsCycles()).toBeTruthy();
+    });
+
+    it('should be able to freeze a specific dependency state', () => {
+        fg.addDependency('1', '2');
+        fg.addDependency('2', '3');
+        fg.freeze();
+        fg.addDependency('3', '4');
+        fg.addDependency('4', '1');
+
+        expect(fg.containsCycles()).toBeTruthy();
+        expect(fg.ids).toEqual(['1', '2', '3', '4']);
+        fg.melt();
+
+        expect(fg.containsCycles()).toBeFalsy();
+        expect(fg.ids).toEqual(['1', '2', '3']);
+
+        fg.melt();
+
+        expect(fg.ids).toEqual(['1', '2', '3']);
+    });
+
+    it('should also being able to handle wildcard dependency as from dependency', () => {
```

Comment:
(this comment applies to other test cases)
I think the word 'also' is unnecessary 
Or shall it indicate that this case is not really intended to be working (optional)?

Reply:
I will change the message. It is an artifact from having written a test for a fixed feature. Therefore the word "also" snugged in.